engine/access/state_stream/backend/registers_async.go

```diff
@@ -0,0 +1,64 @@
+package backend
```

Comment:
this seems fairly general purpose. how about moving it somewhere in `module`. maybe `module/execution/registers.go`